File: 10-stacks-queues1/3-queue-operations/main.cpp
// Queue Operations - C++ Example

#include <iostream>
#include <queue>

int main() {
    std::queue<int> queue;

    queue.push(1);
    queue.push(2);

    if (!queue.empty())
        std::cout << "Front element: " << queue.front() << std::endl;

    if (!queue.empty()) {
        std::cout << "Dequeue element: " << queue.front() << std::endl;
        queue.pop();
    }

    std::cout << "Is queue empty? " << (queue.empty() ? "Yes" : "No") << std::endl;

    return 0;
}

File: 10-stacks-queues1/3-queue-operations/main.py
# Queue Operations - Python Example

from collections import deque

queue = deque()

def enqueue(val):
    queue.append(val)

def dequeue():
    if queue:
        return queue.popleft()
    return None

def front():
    if queue:
        return queue[0]
    return None

def is_empty():
    return len(queue) == 0

enqueue(1)
enqueue(2)
print("Front element:", front())
print("Dequeue element:", dequeue())
print("Is queue empty?", is_empty())
